discovery/validation_barrier.go

```diff
@@ -0,0 +1,29 @@
+package discovery
+
+// ValidationBarrier is an interface that handles dependency issues when
+// processing routing gossip messages defined in BOLT07. As announcements come
+// concurrently, whenever we see a channel_update or a node_announcement, we
+// want to hold processing them until the corresponding channel_announcement is
+// received and processed. This dependency requirement means
+// `lnwire.ChannelAnnouncement` must be processed first before any
+// `lnwire.NodeAnnouncement` or `lnwire.ChannelUpdate` is processed.
+//
+// In addition, when implementing this interface, limiting the number of
+// concurrent jobs should also be considered.
+type ValidationBarrier interface {
+	// InitJobDependencies will wait for a new job slot to become open, and
+	// then sets up any dependent signals/trigger for the new job
+	InitJobDependencies(nMsg *networkMsg)
+
+	// WaitForDependants will block until any jobs that this job dependants
+	// on have finished executing. This allows us a graceful way to
+	// schedule goroutines based on any pending uncompleted dependent jobs.
+	// If this job doesn't have an active dependent, then this function
+	// will return immediately.
+	WaitForDependants(nMsg *networkMsg) error
+
+	// SignalDependants will allow/deny any jobs that are dependent on this
+	// job that they can continue execution. If the job doesn't have any
+	// dependants, then this function sill exit immediately.
+	SignalDependants(nMsg *networkMsg, signal bool)
```

Comment:
Perhaps we should do something like:
```go
type SignalType uint8 

const (
    SignalAllow = iota 
    SignalDeny 
)

SignalDependants(nMsg *networkMsg, signal SignalType)
```

Reply:
Do we plan to support more signals? I'd prefer explicit signals, but atm it seems the bool is ok since we only have two signals.